src/sentry/auth/helper.py

```diff
@@ -226,12 +227,57 @@ def _get_login_redirect(self, subdomain: str | None) -> str:
             login_redirect_url = absolute_uri(login_redirect_url, url_prefix=url_prefix)
         return login_redirect_url
 
-    def _handle_new_membership(self, auth_identity: RpcAuthIdentity) -> RpcOrganizationMember:
-        user, om = auth_service.handle_new_membership(
+    def _handle_membership(
+        self,
+        request: Request,
+        organization: RpcOrganization,
+        auth_identity: AuthIdentity,
+    ) -> Tuple[User, RpcOrganizationMember]:
+        # TODO: Might be able to keep hold of the RpcUser object whose ID was
+        #  originally passed to construct the RpcAuthIdentity object
+        user = User.objects.get(id=auth_identity.user_id)
```

Comment:
I think the comment is out of date, since there's no longer an `RpcAuthIdentity` object (rather an `AuthIdentity` object).

Can you comment on the state of `request.user` at this point in the flow? Would we ever be able to retrieve the value from there instead of getting it from the DB?

Reply:
Yeah, so this is the sticky part of invites:

First off, the auth identity can be a different user than the `request.user` in the case of ambiguous email handling, which gets handled for in the invite helper here:

```
            invite = organization_service.get_invite(
                organization_id=organization_id, email=email, user_id=request.user.id
            )
```

But the point of the user on the request is also ensuring that we don't duplicate users into an org, *EVEN* in the case of multiple emails that might result in doing so.  So this is intended though complicated (and tested).